src/ext/searchbox.js

```diff
@@ -188,7 +189,8 @@ class SearchBox {
                 if (all > MAX_COUNT)
                     break;
                 if (!m[0]) {
-                    regex.lastIndex = last += 1;
+                    regex.lastIndex = last +=
+                        supportsUnicodeFlag && (value.charCodeAt(last) ^ 0xd800) < 1024 ? 2 : 1;
```

Comment:
I see this check repeated 3 times in this PR and it may look like magic to people unfamiliar with bitwise XOR. My understanding is it checks if the character falls into a particular range of unicode characters (range of 10 last bits starting with 0xd800) to determine if it's the "surrogate character", which is unusable on it's own. Let's extract it to a function with a meaningful name and leave a comment there explaining what the function does, this will be helpful in a long term.

Reply:
I did this to avoid using `codePointAt` which has worse browser support. But since only browsers that support the `u` flag will run that code, we can use it. With `codePointAt`, the code could be written like this:

```javascript
regex.lastIndex = last += supportsUnicodeFlag && value.codePointAt(last) > 0xffff ? 2 : 1;
```

Is this readable enough, or should it still be extracted to its own function?

Reply:
I'd prefer it extracted anyway, to avoid unnecessary duplication - it's the same check in all the places. And it will be easier to cover that specific function with tests. I have no strong preference for implementation, so I'll leave it up to you - as long as it works and there are tests in place to prevent a regression :)

Reply:
Got it. Is `lib/lang` a good place to put it?

And lastly, is there a way to run a subset of the tests so I don't have to wait minutes every time I want to test new tests?

Reply:
> Is lib/lang a good place to put it?

I think yes.

> And lastly, is there a way to run a subset of the tests so I don't have to wait minutes every time I want to test new tests?

Try running the tests in the browser by serving http://localhost:8888/src/test/tests.html and selecting a specific test suite there, that is what I used to do for this purpose. Let me know if it doesn't work out.

Reply:
Done. Both of the added tests have been verified to fail on the current version of Ace.

Reply:
Thank you, the PR was merged 🙂